src/sire/system/_system.py

```diff
@@ -304,6 +326,68 @@ def coordinates(self, *args, **kwargs):
         """
         return self.molecules().coordinates(*args, **kwargs)
 
+    def space(self):
+        """
+        Return the space used for this system
+        """
+        try:
+            return self._system.property("space")
+        except Exception:
+            from ..vol import Cartesian
+
+            return Cartesian()
+
+    def time(self):
+        """
+        Return the current system time
+        """
+        try:
+            return self._system.property("time")
+        except Exception:
+            from ..units import picosecond
+
+            return 0 * picosecond
+
+    def set_space(self, space):
+        """
+        Set the space to be used to hold all of the molecules
+        in this system
+        """
+        from ..legacy.Vol import Space
+
+        if not issubclass(type(space), Space):
+            raise TypeError(
+                "You can only set the space to a type derived from sire.vol.Space, "
+                "e.g. sire.vol.PeriodicBox, sire.vol.TriclinicBox or "
+                f"sire.vol.Cartesian. You cannot use a {type(space)}."
+            )
+
+        self._system.set_property("space", space)
+
+    def set_time(self, time):
+        """
+        Set the current time for the system
+        """
+        from ..units import picosecond
+        from ..base import wrap
+
+        if time == 0:
+            self._system.set_property("time", wrap(0 * picosecond))
+        else:
+            if not hasattr(time, "has_same_units"):
+                raise TypeError(
+                    "You can only set the time to a value with units 'time', e.g. "
+                    f"5 * sire.units.picosecond. YOu cannot use a {type(time)}."
+                )
+
+            if not time.has_same_units(picosecond):
+                raise TypeError(
+                    "You can only set the time to a value of units time. You "
+                    f"cannot use {time}."
+                )
+
+            self._system.set_property("time", wrap(time))
```

Comment:
Perhaps you just use default property names in the new API, but I was wondering if we wanted to add an optional `map` parameter so that the user can name or retrieve properties based on a particular mapping?

Reply:
Latest push includes map support :-)